src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/LegacyPackageReferenceProject.cs

```diff
@@ -38,6 +38,8 @@ public sealed class LegacyPackageReferenceProject : BuildIntegratedNuGetProject
         private string _projectName;
         private string _projectUniqueName;
         private string _projectFullPath;
+        private Dictionary<NuGetFramework, List<Utility.InstalledPackageCache>> _installedPackages = new Dictionary<NuGetFramework, List<Utility.InstalledPackageCache>>();
```

Comment:
Given that you for each package you look for the framework, then the package id, I'd recommend the larger collection to be a dictionary for cheaper lookup. 

For a given project, you're likely to have a few frameworks, but you could easily have 20-30 packages.